runtime/caml/domain_state.h

```diff
@@ -0,0 +1,56 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*      KC Sivaramakrishnan, Indian Institute of Technology, Madras       */
+/*                Stephen Dolan, University of Cambridge                  */
+/*                                                                        */
+/*   Copyright 2019 Indian Institute of Technology, Madras                */
+/*   Copyright 2019 University of Cambridge                               */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#ifndef CAML_STATE_H
+#define CAML_STATE_H
+
+#include <stddef.h>
+#include "misc.h"
+#include "mlvalues.h"
+
+#undef young_ptr
+#undef young_limit
+#undef young_start
+#undef young_end
+#undef local_roots
+
+/* This structure sits in the TLS area and is also accessed efficiently
+ * via native code, which is why the indices are important */
+typedef struct {
+#define DOMAIN_STATE(type, name) CAMLalign(8) type name;
```

Comment:
There are a few double values in the domain state (`stat_minor_words`, `stat_promoted_words`, `stat_major_words`, `extra_heap_resources_minor`) which consume 8 bytes even on a 32-bit architecture. It seemed easiest to align everything to 8 bytes rather than work around for doubles.

Reply:
You're right, I didn't think of doubles.